tests/acceptance/lazy-routeable-engine-test.js

```diff
@@ -66,6 +63,40 @@ test('it should pause to load JS and CSS assets on deep link into a lazy Engine'
   andThen(() => verifyInitialBlogRoute(assert, this.loadEvents, this.application));
 });
 
+test('it should pause to load JS and CSS assets on deep link into a lazy Engine after retrying the transition', function(assert) {
+  assert.expect(7);
+
+  define('dummy/routes/application', () => (Ember.Route.extend({
+      actions: {
+        error: nope
+      }
+    })
+  ));
+
+  const adapterException = Ember.Test.adapter.exception;
+  const loader = this.application.__container__.lookup('service:asset-loader');
```

Comment:
I think you can use `this.loader` here (since you saved it off on the test context above in the `beforeEach`).